Persistence/EntityConfigurations/TransactionConfigurations.cs

```diff
@@ -0,0 +1,89 @@
+        {
+            
+            builder.Property(x => x.Amount).IsRequired();
+            builder.Property(x => x.BankAccountId).IsRequired();
+            builder.Property(x => x.DateFiled).IsRequired();
+            builder.Property(x => x.Description).IsRequired();
+            builder.Property(x => x.CounterPartyName).IsRequired();
+            builder.OwnsMany(x => x.Attachments, 
+                AttachmentConfiguration("Transaction_Attachments"));
+            builder.OwnsMany(x => x.TransactionTypeAmounts,
+                TransactionTypeAmountConfiguration("Transaction_TransactionTypeAmounts"));
+
+        }
+
+        private Action<OwnedNavigationBuilder<CreditTransaction, TransactionTypeAmount>> TransactionTypeAmountConfiguration(string tableName)
+        {
+            return cfg =>
+            {
+                cfg.ToTable(tableName);
+                cfg.Property(x => x.Amount).IsRequired();
+                cfg.Property(x => x.TransactionType).IsRequired();
+            };
+        }
+
+        private Action<OwnedNavigationBuilder<CreditTransaction, TransactionAttachment>> AttachmentConfiguration(string tableName)
+        {
+            return cfg =>
+            {
+                cfg.ToTable(tableName);
+                cfg.Property(x => x.Name).IsRequired();
+                cfg.Property(x => x.BlobURI).IsRequired();
+            };
+        }
+    }
+
+    public class DebitTransactionConfiguration : IEntityTypeConfiguration<DebitTransaction>
+    {
+
+        public void Configure(EntityTypeBuilder<DebitTransaction> builder)
+        {
+
+            builder.Property(x => x.Amount).IsRequired();
+            builder.Property(x => x.BankAccountId).IsRequired();
+            builder.Property(x => x.DateFiled).IsRequired();
+            builder.Property(x => x.Description).IsRequired();
+            builder.Property(x => x.CounterPartyName).IsRequired();
+            builder.OwnsMany(x => x.Attachments,
+                AttachmentConfiguration("Transaction_Attachments"));
+            builder.OwnsMany(x => x.TransactionTypeAmounts,
+                TransactionTypeAmountConfiguration("Transaction_TransactionTypeAmounts"));
+
+        }
+
+        private Action<OwnedNavigationBuilder<DebitTransaction, TransactionTypeAmount>> TransactionTypeAmountConfiguration(string tableName)
+        {
+            return cfg =>
+            {
+                cfg.ToTable(tableName);
+                cfg.Property(x => x.Amount).IsRequired();
+                cfg.Property(x => x.TransactionType).IsRequired();
+            };
+        }
+
+        private Action<OwnedNavigationBuilder<DebitTransaction, TransactionAttachment>> AttachmentConfiguration(string tableName)
+        {
+            return cfg =>
+            {
+                cfg.ToTable(tableName);
+                cfg.Property(x => x.Name).IsRequired();
+                cfg.Property(x => x.BlobURI).IsRequired();
+            };
+        }
+    }
+}
```

Comment:
The entity configurations don't have much configuration in them, I'd at least want to see some MaxLength configuration for strgins in here.